lib/ptm_lib.c

```diff
@@ -429,6 +429,7 @@ int ptm_lib_process_msg(ptm_lib_handle_t *hdl, int fd, char *inbuf, int inlen,
 			hdl->response_cb(arg, p_ctxt);
 		break;
 	default:
+		free(p_ctxt);
```

Comment:
What about freeing also?

```c
csv_clean(p_ctxt->csv);
csv_free(p_ctxt->csv);
```

Reply:
Yeah this patch is a no from me.  p_ctxt->type is a enum.  The switch statement already handles all the cases.  This cannot happen.